### PR TITLE
test: Unified card enrollment flow remaining branches code coverage

### PR DESCRIPTION
**Pull request by OmkarJ13**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=XFHgzqU)
### <samp>🤖 Generated by Copilot at 004a3c3</samp>

This pull request improves the testing and code quality of the real-time feed service and the manage corporate cards page. It simplifies the unenrollment of corporate cards by passing the whole card object instead of its parts. It also fixes some minor code style issues with trailing commas and line breaks.

### <samp>🤖 Generated by Copilot at 004a3c3</samp>

> _`unenroll` changes_
> _simplify card management_
> _code style improved too_

### <samp>🤖 Generated by Copilot at 004a3c3</samp>

*  Simplify the `unenroll` method of the `RealTimeFeedService` to accept a `PlatformCorporateCard` object instead of a `cardType` and a `cardId` ([link](https://github.com/fylein/fyle-mobile-app/pull/2398/files?diff=unified&w=0#diff-aea5e7bc6f6a6327858f8ff03ba55f605d40ebd49f410603a103e1e787bae292L55-R62), [link](https://github.com/fylein/fyle-mobile-app/pull/2398/files?diff=unified&w=0#diff-aea5e7bc6f6a6327858f8ff03ba55f605d40ebd49f410603a103e1e787bae292L77-R78))
* Update the `disconnectCard` method of the `ManageCorporateCardsPage` and its test case to use the `card` object instead of the `cardType` and `cardId` when calling the `unenroll` method ([link](https://github.com/fylein/fyle-mobile-app/pull/2398/files?diff=unified&w=0#diff-b389a5d0241bda35125b36240b1d1cc470932c179782063153e1d1ad67705ed8L359-R359), [link](https://github.com/fylein/fyle-mobile-app/pull/2398/files?diff=unified&w=0#diff-450aa28d32fd44fd50bd49190bd3028ca78d9b54b8849b79192b7a2fc10a008fL200-R202))
* Add new test cases for the `getCardType` method of the `RealTimeFeedService` to check if it returns the correct card type based on the card network ([link](https://github.com/fylein/fyle-mobile-app/pull/2398/files?diff=unified&w=0#diff-ca3a1b86f7fac325f1f5f728e37c80dcd0ab178fcdfd9a33ffec734878b4b9b4R55-R68))
* Add new test cases for the `unenroll` method of the `RealTimeFeedService` to check if it handles the unenrollment of visa and mastercard rtf cards and throws an error if the card is not enrolled to rtf ([link](https://github.com/fylein/fyle-mobile-app/pull/2398/files?diff=unified&w=0#diff-ca3a1b86f7fac325f1f5f728e37c80dcd0ab178fcdfd9a33ffec734878b4b9b4R157-R198))
* Add a trailing comma to various method calls for consistency with the code style ([link](https://github.com/fylein/fyle-mobile-app/pull/2398/files?diff=unified&w=0#diff-ca3a1b86f7fac325f1f5f728e37c80dcd0ab178fcdfd9a33ffec734878b4b9b4L29-R29), [link](https://github.com/fylein/fyle-mobile-app/pull/2398/files?diff=unified&w=0#diff-ca3a1b86f7fac325f1f5f728e37c80dcd0ab178fcdfd9a33ffec734878b4b9b4L94-R108), [link](https://github.com/fylein/fyle-mobile-app/pull/2398/files?diff=unified&w=0#diff-ca3a1b86f7fac325f1f5f728e37c80dcd0ab178fcdfd9a33ffec734878b4b9b4L109-R124), [link](https://github.com/fylein/fyle-mobile-app/pull/2398/files?diff=unified&w=0#diff-ca3a1b86f7fac325f1f5f728e37c80dcd0ab178fcdfd9a33ffec734878b4b9b4L119-R133), [link](https://github.com/fylein/fyle-mobile-app/pull/2398/files?diff=unified&w=0#diff-aea5e7bc6f6a6327858f8ff03ba55f605d40ebd49f410603a103e1e787bae292L112-R113), [link](https://github.com/fylein/fyle-mobile-app/pull/2398/files?diff=unified&w=0#diff-450aa28d32fd44fd50bd49190bd3028ca78d9b54b8849b79192b7a2fc10a008fL41-R41), [link](https://github.com/fylein/fyle-mobile-app/pull/2398/files?diff=unified&w=0#diff-450aa28d32fd44fd50bd49190bd3028ca78d9b54b8849b79192b7a2fc10a008fL57-R57), [link](https://github.com/fylein/fyle-mobile-app/pull/2398/files?diff=unified&w=0#diff-450aa28d32fd44fd50bd49190bd3028ca78d9b54b8849b79192b7a2fc10a008fL64-R66), [link](https://github.com/fylein/fyle-mobile-app/pull/2398/files?diff=unified&w=0#diff-450aa28d32fd44fd50bd49190bd3028ca78d9b54b8849b79192b7a2fc10a008fL70-R73), [link](https://github.com/fylein/fyle-mobile-app/pull/2398/files?diff=unified&w=0#diff-450aa28d32fd44fd50bd49190bd3028ca78d9b54b8849b79192b7a2fc10a008fL79-R82), [link](https://github.com/fylein/fyle-mobile-app/pull/2398/files?diff=unified&w=0#diff-450aa28d32fd44fd50bd49190bd3028ca78d9b54b8849b79192b7a2fc10a008fL119-R121), [link](https://github.com/fylein/fyle-mobile-app/pull/2398/files?diff=unified&w=0#diff-450aa28d32fd44fd50bd49190bd3028ca78d9b54b8849b79192b7a2fc10a008fL156-R158))
* Add a line break to various `map` calls for consistency with the code style ([link](https://github.com/fylein/fyle-mobile-app/pull/2398/files?diff=unified&w=0#diff-450aa28d32fd44fd50bd49190bd3028ca78d9b54b8849b79192b7a2fc10a008fL64-R66), [link](https://github.com/fylein/fyle-mobile-app/pull/2398/files?diff=unified&w=0#diff-450aa28d32fd44fd50bd49190bd3028ca78d9b54b8849b79192b7a2fc10a008fL70-R73), [link](https://github.com/fylein/fyle-mobile-app/pull/2398/files?diff=unified&w=0#diff-450aa28d32fd44fd50bd49190bd3028ca78d9b54b8849b79192b7a2fc10a008fL79-R82), [link](https://github.com/fylein/fyle-mobile-app/pull/2398/files?diff=unified&w=0#diff-450aa28d32fd44fd50bd49190bd3028ca78d9b54b8849b79192b7a2fc10a008fL119-R121))
* Import a new mock data object `statementUploadedCard` for testing the `RealTimeFeedService` ([link](https://github.com/fylein/fyle-mobile-app/pull/2398/files?diff=unified&w=0#diff-ca3a1b86f7fac325f1f5f728e37c80dcd0ab178fcdfd9a33ffec734878b4b9b4L5-R5))

## Clickup
https://app.clickup.com

### CODE REVIEW

File: src/app/core/services/corporate-credit-card-expense.service.spec.ts [hunk 1]
```diff
@@ -139,6 +140,15 @@ describe('CorporateCreditCardExpenseService', () => {
     expect(result).toEqual(expectedUniqueCardStats);
   });
 
+  it('getPlatformCorporateCardDetails(): should get expense details in platform card', () => {
```
Comment (Chethan-Fyle) [inline src/app/core/services/corporate-credit-card-expense.service.spec.ts:143]: The function name and the description are not in sync. What is `platform card`?

File: src/app/core/services/real-time-feed.service.spec.ts [hunk 2]
```diff
@@ -140,6 +154,48 @@ describe('RealTimeFeedService', () => {
     }));
   });
 
+  describe('unenroll()', () => {
+    it('should handle unenrollment of visa rtf cards', () => {
```
Comment (Chethan-Fyle) [inline src/app/core/services/real-time-feed.service.spec.ts:158]: `should unenroll visa rtf card when VISA card is passed`

**Review by Chethan-Fyle — APPROVED** [review]
minor comment.

File: src/app/core/services/corporate-credit-card-expense.service.spec.ts (hunk 1 above)
Comment (OmkarJ13) [reply src/app/core/services/corporate-credit-card-expense.service.spec.ts]: rephrased this to `should get corporate card details`

**Pull request merged**